src/components/Checkout/CheckoutDropInPaymentWrapper.vue

```diff
@@ -339,39 +340,61 @@ export default {
 					},
 				})
 				.then(kivaBraintreeResponse => {
+					// extract transaction saga id or transaction id from response
+					const transactionResult = this.useAsyncCheckout
+						? kivaBraintreeResponse?.data?.shop?.doNoncePaymentDepositAndCheckoutAsync
+						: kivaBraintreeResponse?.data?.shop?.doNoncePaymentDepositAndCheckout;
+
+					if (this.useAsyncCheckout && typeof transactionResult !== 'object') {
+						pollForCheckoutStatus(this.apollo, transactionResult)
+							.then(checkoutStatusResponse => {
+								this.handleSuccessfulCheckout(checkoutStatusResponse?.receipt?.checkoutId);
+							}).catch(errorResponse => {
+								this.handleFailedCheckout([
+									{
+										error: errorResponse.errorCode,
+										message: `${errorResponse?.errorMessage}, ${errorResponse?.status}`,
+									}
+								]);
+							});
+
+						return kivaBraintreeResponse;
+					}
+
 					// Check for errors in transaction
 					if (kivaBraintreeResponse.errors) {
-						this.$emit('updating-totals', false);
-						this.processBraintreeDropInError('basket', kivaBraintreeResponse);
-						// Payment method failed, unselect attempted payment method
-						this.$refs.braintreeDropInInterface.btDropinInstance.clearSelectedPaymentMethod();
-						// Initialize a refresh of basket state
-						this.$emit('refreshtotals');
-						// exit
+						this.handleFailedCheckout(kivaBraintreeResponse);
 						return kivaBraintreeResponse;
 					}
 
-					// Transaction is complete
-					const transactionId = _get(
-						kivaBraintreeResponse,
-						'data.shop.doNoncePaymentDepositAndCheckout'
-					);
-					// redirect to thanks with KIVA transaction id
-					if (transactionId) {
-						// fire BT Success event
-						this.$kvTrackEvent(
-							'basket',
-							`${paymentType} Braintree DropIn Payment`,
-							'Success',
-							transactionId,
-							transactionId
-						);
-						// Complete transaction handles additional analytics + redirect
-						this.$emit('complete-transaction', transactionId);
-					}
+					this.handleSuccessfulCheckout(transactionResult, paymentType);
```

Comment:
Should this code be in an `else`? It looks like the `if` above doesn't `await`, so this code will be hit while the async checkout is polling.

Reply:
The if block above doesn't contain any async operations and should fire the various tracking/logging methods and return if there are errors.

Reply:
I meant the `if` on line 348. It calls the async checkout and has `handleSuccessfulCheckout` in the callback. But this line will also call `handleSuccessfulCheckout`, since there is no `else`.

Reply:
Oh never mind, I missed the `return`. Resolving this comment.